experimental/GraphicalModels/src/PhylogeneticAuxiliary.jl

```diff
@@ -0,0 +1,95 @@
+#### GROUP OPERATIONS #### 
+
+# This function implements the group operation on Z2 x Z2 and is subsequently 
+# used to check the condition given in the computation of the Fourier
+# coordinates and to calculate the sum of the leaves beneath a given edge. 
+function group_sum(pm::PhylogeneticModel, states::Vector{Int})
+    group = group_model(pm)
+  
+    if (length(states) == 1)
+      return(group[states[1]])
```

Comment:
Return is not a function, so please don't write `return(FOO)` here or elsewhere.
```suggestion
      return group[states[1]]
```